src/raftstore/store/peer.rs

```diff
@@ -1606,7 +1673,12 @@ impl Peer {
         }
 
         if req.get_admin_request().has_prepare_merge() {
-            self.pre_propose_merge(req)?
+            self.pre_propose_prepare_merge(req)?;
+            ctx.insert(ProposalContext::PREPARE_MERGE);
```

Comment:
I think you should add `ProposalContext::MERGE` when `req.get_admin_request().has_merge()`.